src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7188,7 +7188,18 @@ private BoundExpression BindIndexerAccess(ExpressionSyntax node, BoundExpression
 
             if (!lookupResult.IsMultiViable)
             {
-                indexerAccessExpression = BadIndexerExpression(node, expr, analyzedArguments, lookupResult.Error, diagnostics);
+                if (TryBindIndexOrRangeIndexer(
+                    node,
+                    expr,
```

Comment:
We commonly style indexer access expressions as `receiver[expr]`, so naming this `expr` here seems confusing.

Reply:
Actually, indexers can have multiple arguments, so it would be `receiver[arg1, arg2, arg3]` so using `expr` as the argument name just means we're inserting an expression (which could be any of those components).